### PR TITLE
Remove placeholder constants

### PR DESCRIPTION
**Pull request by Lilit0x**
This PR removes placeholder constants as described in #3131. Below are lists of patterns matched for unused constant names. 
It also includes an update of the `IPROTO_MAX` constant on newer versions of Linux mentioned in #1896. 
- `_COUNT`
- `_MAX`
- `_NUM`

These are all breaking changes for the new `libc 1.0` version tracked in #3248

### CODE REVIEW

File: src/unix/linux_like/linux/mod.rs [hunk 1]
```diff
@@ -5007,7 +4990,6 @@ pub const VM_HUGETLB_PAGES: ::c_int = 18;
 pub const VM_SWAPPINESS: ::c_int = 19;
 pub const VM_LOWMEM_RESERVE_RATIO: ::c_int = 20;
 pub const VM_MIN_FREE_KBYTES: ::c_int = 21;
-pub const VM_MAX_MAP_COUNT: ::c_int = 22;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:5010]: This is not a placeholder constant: it is an input to a function which returns a count.

File: src/unix/linux_like/linux/mod.rs [hunk 2]
```diff
@@ -3676,7 +3660,6 @@ pub const IW_MICFAILURE_KEY_ID: ::c_ulong = 0x00000003;
 pub const IW_MICFAILURE_GROUP: ::c_ulong = 0x00000004;
 pub const IW_MICFAILURE_PAIRWISE: ::c_ulong = 0x00000008;
 pub const IW_MICFAILURE_STAKEY: ::c_ulong = 0x00000010;
-pub const IW_MICFAILURE_COUNT: ::c_ulong = 0x00000060;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:3679]: Same here.

File: src/unix/linux_like/linux/mod.rs [hunk 3]
```diff
@@ -2463,13 +2458,7 @@ pub const RENAME_WHITEOUT: ::c_uint = 4;
...
 // NOTE: These are in addition to the constants defined in src/unix/mod.rs
 
-#[deprecated(
-    since = "0.2.80",
-    note = "This value was increased in the newer kernel \
-            and we'll change this following upstream in the future release. \
-            See #1896 for more info."
-)]
-pub const IPPROTO_MAX: ::c_int = 256;
+pub const IPPROTO_MAX: ::c_int = 263;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:2461]: This should be removed.

File: src/unix/linux_like/linux/mod.rs [hunk 4]
```diff
@@ -2324,8 +2321,6 @@ pub const IFLA_EVENT: ::c_ushort = 44;
 pub const IFLA_NEW_NETNSID: ::c_ushort = 45;
 pub const IFLA_IF_NETNSID: ::c_ushort = 46;
 pub const IFLA_TARGET_NETNSID: ::c_ushort = IFLA_IF_NETNSID;
-pub const IFLA_CARRIER_UP_COUNT: ::c_ushort = 47;
-pub const IFLA_CARRIER_DOWN_COUNT: ::c_ushort = 48;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:2328]: Not a placeholder.

File: src/unix/linux_like/linux/mod.rs [hunk 5]
```diff
@@ -2307,8 +2306,6 @@ pub const IFLA_GROUP: ::c_ushort = 27;
 pub const IFLA_NET_NS_FD: ::c_ushort = 28;
 pub const IFLA_EXT_MASK: ::c_ushort = 29;
 pub const IFLA_PROMISCUITY: ::c_ushort = 30;
-pub const IFLA_NUM_TX_QUEUES: ::c_ushort = 31;
-pub const IFLA_NUM_RX_QUEUES: ::c_ushort = 32;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:2311]: Not a placeholder.

File: src/unix/linux_like/linux/mod.rs [hunk 6]
```diff
@@ -2297,7 +2297,6 @@ pub const IFLA_LINKMODE: ::c_ushort = 17;
 pub const IFLA_LINKINFO: ::c_ushort = 18;
 pub const IFLA_NET_NS_PID: ::c_ushort = 19;
 pub const IFLA_IFALIAS: ::c_ushort = 20;
-pub const IFLA_NUM_VF: ::c_ushort = 21;
```
Comment (Amanieu) [inline src/unix/linux_like/linux/mod.rs:2300]: Not a placeholder.

File: src/unix/hurd/mod.rs [hunk 7]
```diff
@@ -3375,7 +3375,6 @@ pub const TCP_FIN_WAIT2: ::c_uint = 5;
 pub const TCP_TIME_WAIT: ::c_uint = 6;
 pub const TCP_CLOSE: ::c_uint = 7;
 pub const TCP_CLOSE_WAIT: ::c_uint = 8;
-pub const TCP_LAST_ACK: ::c_uint = 9;
```
Comment (Amanieu) [inline src/unix/hurd/mod.rs:3378]: Not a placeholder.

File: src/unix/haiku/native.rs [hunk 8]
```diff
@@ -835,7 +834,6 @@ pub const B_MEDIA_BAD_CLIP_FORMAT: status_t = B_MEDIA_ERROR_BASE + 121;
 pub const B_MEDIA_ADDON_FAILED: status_t = B_MEDIA_ERROR_BASE + 122;
 pub const B_MEDIA_ADDON_DISABLED: status_t = B_MEDIA_ERROR_BASE + 123;
 pub const B_MEDIA_CHANGE_IN_PROGRESS: status_t = B_MEDIA_ERROR_BASE + 124;
-pub const B_MEDIA_STALE_CHANGE_COUNT: status_t = B_MEDIA_ERROR_BASE + 125;
```
Comment (Amanieu) [inline src/unix/haiku/native.rs:838]: Not a placeholder.

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 9]
```diff
@@ -1757,7 +1756,6 @@ pub const DCCP_MAXSEG: ::c_int = 4;
 pub const DCCP_SERVICE: ::c_int = 8;
 
 pub const DCCP_NDP_LIMIT: ::c_int = 16;
-pub const DCCP_SEQ_NUM_LIMIT: ::c_int = 16777216;
```
Comment (Amanieu) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:1760]: Not a placeholder.

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 10]
```diff
@@ -1730,7 +1730,6 @@ pub const DCCP_OPT_CONFIRM_L: ::c_int = 33;
 pub const DCCP_OPT_CHANGE_R: ::c_int = 34;
 pub const DCCP_OPT_CONFIRM_R: ::c_int = 35;
 pub const DCCP_OPT_INIT_COOKIE: ::c_int = 36;
-pub const DCCP_OPT_NDP_COUNT: ::c_int = 37;
```
Comment (Amanieu) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:1733]: Not a placeholder.

File: src/unix/bsd/freebsdlike/freebsd/mod.rs [hunk 11]
```diff
@@ -3680,7 +3680,6 @@ pub const DEAD_PROCESS: ::c_short = 7;
 pub const SHUTDOWN_TIME: ::c_short = 8;
 // utmp database types
 pub const UTXDB_ACTIVE: ::c_int = 0;
-pub const UTXDB_LASTLOGIN: ::c_int = 1;
```
Comment (Amanieu) [inline src/unix/bsd/freebsdlike/freebsd/mod.rs:3683]: Not a placeholder.

File: src/unix/bsd/freebsdlike/dragonfly/mod.rs [hunk 12]
```diff
@@ -1436,7 +1435,6 @@ pub const DOWNTIME: ::c_short = 11;
 // utmpx database types
 pub const UTX_DB_UTMPX: ::c_uint = 0;
 pub const UTX_DB_WTMPX: ::c_uint = 1;
-pub const UTX_DB_LASTLOG: ::c_uint = 2;
```
Comment (Amanieu) [inline src/unix/bsd/freebsdlike/dragonfly/mod.rs:1439]: Not a placeholder.

File: src/unix/bsd/freebsdlike/dragonfly/mod.rs [hunk 13]
```diff
@@ -1064,7 +1064,6 @@ pub const CPUCTL_CPUID: ::c_int = 0xc0106303;
 pub const CPUCTL_UPDATE: ::c_int = 0xc0106304;
 pub const CPUCTL_MSRSBIT: ::c_int = 0xc0106305;
 pub const CPUCTL_MSRCBIT: ::c_int = 0xc0106306;
-pub const CPUCTL_CPUID_COUNT: ::c_int = 0xc0106307;
```
Comment (Amanieu) [inline src/unix/bsd/freebsdlike/dragonfly/mod.rs:1067]: Not a placeholder.

File: src/unix/bsd/apple/mod.rs [hunk 14]
```diff
@@ -5225,7 +5221,6 @@ pub const VOL_CAP_FMT_DECMPFS_COMPRESSION: attrgroup_t = 0x00010000;
 pub const VOL_CAP_FMT_64BIT_OBJECT_IDS: attrgroup_t = 0x00020000;
 pub const VOL_CAP_FMT_DIR_HARDLINKS: attrgroup_t = 0x00040000;
 pub const VOL_CAP_FMT_DOCUMENT_ID: attrgroup_t = 0x00080000;
-pub const VOL_CAP_FMT_WRITE_GENERATION_COUNT: attrgroup_t = 0x00100000;
```
Comment (Amanieu) [inline src/unix/bsd/apple/mod.rs:5228]: Not a placeholder.

File: src/unix/bsd/apple/mod.rs [hunk 15]
```diff
@@ -5140,7 +5137,6 @@ pub const ATTR_CMN_OWNERID: attrgroup_t = 0x00008000;
 pub const ATTR_CMN_GRPID: attrgroup_t = 0x00010000;
 pub const ATTR_CMN_ACCESSMASK: attrgroup_t = 0x00020000;
 pub const ATTR_CMN_FLAGS: attrgroup_t = 0x00040000;
-pub const ATTR_CMN_GEN_COUNT: attrgroup_t = 0x00080000;
```
Comment (Amanieu) [inline src/unix/bsd/apple/mod.rs:5143]: Not a placeholder.

File: src/unix/aix/mod.rs [hunk 16]
```diff
@@ -1404,8 +1404,6 @@ pub const LPAR_INFO_FORMAT1: ::c_int = 1;
 pub const LPAR_INFO_FORMAT2: ::c_int = 2;
 pub const WPAR_INFO_FORMAT: ::c_int = 3;
 pub const PROC_MODULE_INFO: ::c_int = 4;
-pub const NUM_PROC_MODULE_TYPES: ::c_int = 5;
-pub const LPAR_INFO_VRME_NUM_POOLS: ::c_int = 6;
```
Comment (Amanieu) [inline src/unix/aix/mod.rs:1408]: Not a placeholder.

**Review by Amanieu — COMMENTED** [review]
Many of these are not placeholder constants, you should individually double-check each of them.

**Comment by JohnTitor** [comment]
Closing as inactive, it'd be great if you could split PRs to make our review easy. Thank you for the PR anyway!

**Comment by Lilit0x** [comment]
> Closing as inactive, it'd be great if you could split PRs to make our review easy. Thank you for the PR anyway!

Yeahh, I paused on it because some clarification is needed based on my discussion with @Amanieu .

**Pull request closed without merging**